src/dbnode/namespace/schema.go

```diff
@@ -278,20 +297,43 @@ func marshalFileDescriptors(fdList []*desc.FileDescriptor) ([][]byte, error) {
 	return dlist, nil
 }
 
-func LoadSchemaRegistryFromFile(schemaReg SchemaRegistry, nsID ident.ID, protoFile string, msgName string, importPathPrefix ...string) error {
-	out, _ := parseProto(protoFile, importPathPrefix...)
+func AppendSchemaOptions(schemaOpt *nsproto.SchemaOptions, protoFile, msgName string, contents map[string]string, deployID string, importPath ...string) (*nsproto.SchemaOptions, error) {
```

Comment:
So just to clarify: The way this work is that you provide the `contents` which is a map[string]string of all of the filenames to their contents. `msgName` is the name of the message that you want to use as the schema and `protoFile` is the name of the file that contains `msgName`?

If so, could you add a comment to the top of this method explaining all that. I was able to work it out cause I have context here, but I think it would be really confusing to decipher on your own

Reply:
Does the `importPath` argument matter at all? I looked at this: https://godoc.org/github.com/jhump/protoreflect/desc/protoparse#Parser

and it doesn't seem relevant to our use-case (and it doesn't seem like you're using it from the service layer?)

Reply:
I did use this when loading from file. but I can certainly separate them so that they dont' share the same code, it might be clearer.